peps/pep-0784.rst

```diff
@@ -0,0 +1,277 @@
+PEP: 784
+Title: Adding Zstandard to the standard library
+Author: Emma Harper Smith <emma@python.org>
+Sponsor: Gregory P. Smith <greg@krypto.org>
+Status: Draft
+Type: Standards Track
+Created: 04-Apr-2025
+Python-Version: 3.15
+
+Abstract
+========
+
+`Zstandard <https://facebook.github.io/zstd/>`_ is a widely adopted, mature,
+and highly efficient compression standard. This PEP proposes adding a new
+module to the Python standard library containing a Python wrapper around Meta's
+``zstd`` library, the default implementation. Additionally, to avoid name
+collisions with packages on PyPI and to present a unified interface to Python
+users, compression modules in the standard library will be moved under a
+``compression.*`` namespace package.
+
+Motivation
+==========
+
+CPython has modules for several different compression formats, such as `zlib
+(DEFLATE) <https://docs.python.org/3/library/zlib.html>`_,
+`bzip2 <https://docs.python.org/3/library/bz2.html>`_,
+and `lzma <https://docs.python.org/3/library/lzma.html>`_, each widely used.
+Including popular compression algorithms matches Python's "batteries included"
+philosophy of including widely useful standards and utilities. The last
+compression module added to the language was ``lzma``, added in Python 3.3.
+Since that time, several new compression formats have become very popular,
+including Zstandard. Zstandard uses highly optimized implementations of modern
+compression techniques such as `Asymmetric Numerical Systems (ANS)
+<https://en.wikipedia.org/wiki/Asymmetric_numeral_systems>`_
+to acheive high compression ratios, while still attaining high performance.
+
+Zstandard was released over a decade ago. In the intervening time, it has seen
+`widespread adoption in many different areas of computing <https://facebook.github.io/zstd/#references>`_,
+including databases, filesystems, archive formats (including ``tar`` and
+``zip``), package formats (including conda packages), and other file formats
+(including several Apache formats like Arrow).
+
+There are several bindings to Zstandard for Python available on PyPI, each with
+different APIs and choices of how to bind the ``zstd`` library. One goal with
+introducing an official module in the standard library is to reduce confusion
+for Python users who want simple compression/decompression APIs for Zstandard.
+The existing packages can continue providing extended APIs and bindings for
+other Python implementations such as PyPy or integrate features from newer
+Zstandard versions.
+
+Another reason to add Zstandard support to the standard library is to resolve
```

Comment:
I'd add a note at the end of this paragraph that the `zipfile` module also needs zstd support.  https://github.com/facebook/zstd/issues/1378#issuecomment-2129654131 documents how.  it's just a compression format id.